### PR TITLE
ENG-199 Signup Screens (Static)

### PR DESCRIPTION
**Pull request by HoseaCodes**

### CODE REVIEW

File: src/components/auth/AuthComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      case 1:
+        return <SignupComponent screen={screen} nextStep={nextStep} handleInputChange={handleInputChange} navigation={navigation} />;
+      case 2:
+        return <PhoneLoginScreen nextStep={nextStep} prevStep={prevStep} handleInputChange={handleInputChange} navigation={navigation} />;
+      case 3:
+        return <MFAScreen nextStep={nextStep} prevStep={prevStep} handleInputChange={handleInputChange} navigation={navigation} route={route} />;
+      default:
+        return null;
+    }
+  }
```
Comment (rfougy) [inline src/components/auth/AuthComponent.tsx:75]: There's a possibility that `renderLoginStep` can be consolidated into renderSignUpStep:

```
  const renderAuthSteps = () => {
    switch (step) {
      case 1:
        return <SignupComponent screen={screen} nextStep={nextStep} handleInputChange={handleInputChange} navigation={navigation} />;
      case 2:
        return <PhoneSignupScreen nextStep={nextStep} prevStep={prevStep} handleInputChange={handleInputChange} navigation={navigation} />;
      case 3:
        return <MFAScreen nextStep={nextStep} prevStep={prevStep} handleInputChange={handleInputChange} navigation={navigation} route={route} />;
      case 4:
        return screen === 'Signup' && <ChangePassowrdScreen nextStep={nextStep} prevStep={prevStep} handleInputChange={handleInputChange} navigation={navigation} />;
      case 5:
        return screen === 'Signup' && <SignupDetailsScreen nextStep={nextStep} prevStep={prevStep} formData={formData} handleInputChange={handleInputChange} navigation={navigation} />;
      case 6:
        return screen === 'Signup' && <WelcomeScreen nextStep={nextStep} prevStep={prevStep} navigation={navigation} />;
      default:
        return null;
    }
  };
```

File: src/components/auth/AuthComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  return (
+    <View style={styles.container}>
+      {
+        screen === 'Signup' ? renderSignUpStep() : renderLoginStep()
+      }
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (rfougy) [inline src/components/auth/AuthComponent.tsx:86]: It seems that only the `container` styles are being actively used, in which the remainder could be safely removed

File: src/screens/auth/ChangePassowrdScreen.js
Comment (rfougy) [inline src/screens/auth/ChangePassowrdScreen.js:1]: Nitpick: There's a small typo with the filename "Passowrd"

File: src/screens/auth/ChangePassowrdScreen.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, TouchableOpacity, StyleSheet } from 'react-native';
+import Header from '../../components/Header';
+import AntDesign from '@expo/vector-icons/AntDesign';
+import Progressbar from '../../components/Progressbar';
+import { Ionicons } from '@expo/vector-icons'; 
+import ContinueButton from '../../components/ContinueButton';
+
+const ChangePassowrdScreen = ({ nextStep, prevStep, handleInputChange, navigation }) => {
```
Comment (rfougy) [inline src/screens/auth/ChangePassowrdScreen.js:9]: Nitpick: There's a small typo "Passowrd"

File: src/screens/auth/ChangePassowrdScreen.js
Comment (rfougy) [reply src/screens/auth/ChangePassowrdScreen.js:1]: Validation is missing for the requirement of including the following characters: !, #, &, $, @.

File: src/screens/auth/MFAScreen.js [hunk 4]
```diff
@@ -65,10 +76,8 @@ const MFAScreen = ({ navigation }) => {
                     <Text style={styles.timerText}>50s Remaining</Text>
                 </View>
             </View>
-
-            <TouchableOpacity style={styles.continueButton}>
-                <Text style={styles.continueButtonText}>Continue</Text>
-            </TouchableOpacity>
+            { screen == 'signup' && ( <Progressbar activeSteps={2} /> ) }
```
Comment (rfougy) [inline src/screens/auth/MFAScreen.js:79]: Is there a reason for choosing `==` over `===`? From what I understand `===` is more reliable and less error-prone.

**Review by rfougy — CHANGES_REQUESTED** [review]
I have a few notes from QA'ing this ticket:

- [ ] **1. Absence of a Disabled Button UI:** I did not see this in the Figma file and have not reached out Caleb on this, but I'd imagine we would need to style the button differently when disabled? For example having it dulled out in comparison to when a button is readily clickable.
- [ ] **2. No Validation Present:** Validation is not enforced for all fields and the user is able to click the continue button when 1) the fields are incorrect, and 2) the fields are empty. Validation error messages also dissapear after passing a certain character count, despite issues with the data (e.g., incorrect formatting) remaining.
- [ ] **3. Validation UI and Copy not Present in Figma:** There are certain areas in the Figma file that provides examples of validation styling such as [here](https://www.figma.com/design/UfDbJIiBaEuami1unNO8Kc/Mun-e-App-Design?node-id=17-9463&node-type=frame&t=tPwJHJXzupvzyqkL-0). However, it's unclear of this PR's completion of validation UI and copy based off what's currently in the Figma.
- [ ] **4. Countdown Not Occurring:** The time displayed in the MFA screen is not counting down. There also isn’t a wireframe in Figma for the scenario where the countdown expires.
- [ ] **5. Simultaneous Typing in Input Fields:**

https://github.com/user-attachments/assets/08ae3fed-9714-4666-8dab-1d54397e3919

Nitpicks:
- [ ] **6. Unclickable Resend Code:** The 'Resend Code?' text button is not a clickable component.
- [x] **7. Low-res Mun-e Logo:** The Mun-e Logo featured on the Welcome Screen should be an SVG, not a PNG. If using an SVG requires a large amount of time, resorting to png with higher resolution
- [ ] **8. Tapping UX for Login Screen:** When tapping into the input field on a mobile device, I can’t exit by tapping outside of the field and native touchscreen keyboard.

File: src/screens/auth/ChangePassowrdScreen.js
Comment (HoseaCodes) [reply src/screens/auth/ChangePassowrdScreen.js:1]: This was a static only update with no validation

**Pull request merged**